petastorm/spark/spark_dataset_converter.py

```diff
@@ -592,18 +592,23 @@ def wait_for_file(path):
 
 def _check_dataset_file_median_size(url_list):
     fs, path_list = get_filesystem_and_path_or_paths(url_list)
+    SIZE_LIMIT_BYTES = 50 * 1024 * 1024
+    MB_BYTES = 1024 * 1024
 
     # TODO: also check file size for other file system.
     if isinstance(fs, LocalFileSystem):
         pool = ThreadPool(64)
         try:
             file_size_list = pool.map(os.path.getsize, path_list)
-            mid_index = len(file_size_list) // 2
-            median_size = sorted(file_size_list)[mid_index]
-            if median_size < 50 * 1024 * 1024:
-                logger.warning('The median size (%d) of these parquet files (%s) is too small.'
-                               'Increase file sizes by repartition or coalesce spark dataframe, which '
-                               'will help improve performance.', median_size, ','.join(url_list))
+            total_size = sum(file_size_list)
+            if total_size > SIZE_LIMIT_BYTES:
+                mid_index = len(file_size_list) // 2
+                median_size = sorted(file_size_list)[mid_index]
+                if median_size < SIZE_LIMIT_BYTES:
```

Comment:
Making up 3 different thresholds seems pretty complex and arbitrary... Maybe we just keep it simple for now and only use `SIZE_LIMIT_BYTES`. It could also be easier for users to figure out how to get rid of this message: by reducing the numPartitions as much as possible until the median size reaches 50 MB.